src/preferences/dialog/dlgpreferences.cpp

```diff
@@ -371,12 +373,29 @@ void DlgPreferences::onShow() {
         screenSpace = QSize(800, 600);
     }
     else {
-        screenSpace = pScreen->size();
+        screenSpace = pScreen->availableSize();
     }
-    newX = std::max(0, std::min(newX, screenSpace.width() - m_geometry[2].toInt()));
-    newY = std::max(0, std::min(newY, screenSpace.height() - m_geometry[3].toInt()));
+
+    // Make sure the entire window is visible on screen and is not occluded by taskbar
+    // Note: Window geometry excludes window decoration
+    int windowDecorationWidth = frameGeometry().width() - geometry().width();
+    int windowDecorationHeight = frameGeometry().height() - geometry().height();
+    if (windowDecorationWidth <= 0) {
+        windowDecorationWidth = 2;
+    }
+    if (windowDecorationHeight <= 0) {
+        windowDecorationHeight = 30;
+    }
```

Comment:
this does not make sense to me. Would you mind giving some explanation?

Reply:
I'm currently debugging a similiar issue with DlgDevTools¹ and I ended up here, too:
those magic numbers are used as default window decoration dimensions in [QRect DlgPreferences::getDefaultGeometry()](https://github.com/mixxxdj/mixxx/blob/3df7942a0f958e41ebb976002e15ce1ed13481f5/src/preferences/dialog/dlgpreferences.cpp#L534-L557) in case they cannot be retireved from the OS. !? 

¹when I open it the initial window it's like 10.000 pixels wide